### DISCUSSION TITLE
How does the gas usage of PreCommitSectorBatch scale with the number of sectors committed?

### DISCUSSION THREAD

**Discussion by The-Wayvy** [filecoin-project/FIPs#736] [General]
How does the gas usage of PreCommitSectorBatch scale with the number of sectors committed?

Is it logarithmic in the number of sectors?

**Comment by Kubuxu** [comment]
Asymptotically it is still linear, but there is amortisation of flat overhead the more sectors there are.

**Reply by The-Wayvy** [reply]
I tried eyeballing the marginal cost for adding 1 sector to PreCommitSectorBatch on the block explorer.

3,000,000 for each new sector
1/10th of the function overhead

Is that about right? I'm just looking for a rough estimate.

**Comment by anorth** [comment]
We could improve this with non-interactive PoRep (#603), which gets rid of PreCommit altogether!
